File: Card_Game_Engine/Models/Filter/Classes/FilterGroup.cs
using Card_Game_Engine.Models.Filter.Enums;
using Card_Game_Engine.Models.Filter.Interfaces;

namespace Card_Game_Engine.Models.Filter.Classes;

public class FilterGroup : IFilterNode
{
    public List<IFilterNode> Children { get; set; } = new List<IFilterNode>();
    public LogicalOperatorEnum? LogicalOperator { get; set; }
}

File: Card_Game_Engine/ClientApp/src/app/_reusable-components/parameter/filter-modal/_namespace/classes/filter-group.ts
import {IFilterNode} from "../interfaces/filter-node";
import {LogicalOperatorEnum} from "../enums/logical-operator.enum";
import {FilterCondition} from "./filter-condition";

export class FilterGroup implements IFilterNode {
  logicalOperator?: LogicalOperatorEnum;
  children: (FilterGroup | FilterCondition)[];

  constructor(children: (FilterGroup | FilterCondition)[] = [], logicalOperator: LogicalOperatorEnum = null) {
    this.logicalOperator = logicalOperator;
    this.children = children;
  }
}
